packages/server/src/core/parser.rs

```diff
@@ -138,64 +138,58 @@ fn set_grades_for_uncompleted_courses(
     }
 }
 
-fn parse_course_status_pdf_format(line: &str) -> Result<(Course, Option<Grade>), AppError> {
-    let clean_line = line.replace('*', "");
-    let id = {
-        let number = clean_line
-            .split(' ')
-            .next()
-            .ok_or_else(|| AppError::Parser("Bad Format".into()))?;
-        if number.parse::<f32>().is_ok() {
-            Ok(String::from(number))
-        } else {
-            Err(AppError::Parser("Bad Format".into()))
-        }?
-    };
-
-    let mut index = 0;
-    let mut credit = 0.0;
-    let mut word;
-    for part in clean_line.split(' ') {
-        word = part.to_string();
-        // When a grade is missing, a hyphen (מקף) char is written instead, without any whitespaces between it and the credit.
-        // This means that the credit part is no longer parsable as f32, and therefore the hyphen must be manually removed.
-        // This won't create a problem later in the code since 'word' only lives in the for-loop scope.
-        if word.contains('-') && word.contains('.') {
-            word = word.replace('-', "").trim().to_string();
-        }
-        if word.parse::<f32>().is_ok() && word.contains('.') {
-            credit = word
-                .chars()
-                .rev()
-                .collect::<String>()
-                .parse::<f32>()
-                .map_err(|e| AppError::Parser(format!("Bad Format: {e}")))?;
-            break;
-        }
-        index += 1;
-    }
-
-    let name = clean_line.split_whitespace().collect::<Vec<&str>>()[1..index].join(" ");
+fn extract_str_by_regex(
+    line: &str,
+    regex: &Regex,
+    regex_name: &str,
+) -> Result<(String, String), AppError> {
+    let extracted = regex
+        .captures(line)
+        .ok_or_else(|| AppError::Parser("Bad Format".into()))?[regex_name]
+        .trim()
+        .to_string();
+    let line = regex.replace(line, "");
```

Comment:
Why do we need this line?

Reply:
Clearing the match helps match other regexes in some formats (mostly those with less spaces like firefox)

Reply:
I see, so don't you prefer to use replace_all?
Or we assume that every regex appears only once so it doesn't matter?
Anyway I am fine with both solutions.

Reply:
Since this function parses a single line which contains only course name, id, credit and grade then yes, we indeed assume that every regex appears only once